hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -471,134 +558,231 @@ public HashSet<String> readRocksDBLiveFiles(String dbPathArg) {
-      Snapshot src, Snapshot dest,
-      HashSet<String> srcSnapFiles,
-      HashSet<String> destSnapFiles,
+  private void internalGetSSTDiffList(Snapshot src, Snapshot dest,
+      HashSet<String> srcSnapFiles, HashSet<String> destSnapFiles,
       MutableGraph<CompactionNode> mutableGraph,
       HashSet<String> sameFiles, HashSet<String> differentFiles) {
 
-
     for (String fileName : srcSnapFiles) {
       if (destSnapFiles.contains(fileName)) {
-        LOG.warn("SrcSnapshot : " + src.dbPath + " and Dest " +
-            "Snapshot" + dest.dbPath + " Contain Same file " + fileName);
+        LOG.debug("Source '{}' and destination '{}' share the same SST '{}'",
+            src.dbPath, dest.dbPath, fileName);
         sameFiles.add(fileName);
         continue;
       }
       CompactionNode infileNode =
           compactionNodeTable.get(Paths.get(fileName).getFileName().toString());
       if (infileNode == null) {
-        LOG.warn("SrcSnapshot : " + src.dbPath + "File " + fileName + "was " +
-            "never compacted");
+        LOG.debug("Src " + src.dbPath + " File " + fileName +
+            " was never compacted");
         differentFiles.add(fileName);
         continue;
       }
-      System.out.print(" Expandin File:" + fileName + ":\n");
-      Set<CompactionNode> nextLevel = new HashSet<>();
-      nextLevel.add(infileNode);
+      LOG.debug("Expanding SST file: " + fileName);
       Set<CompactionNode> currentLevel = new HashSet<>();
-      currentLevel.addAll(nextLevel);
-      nextLevel = new HashSet<>();
+      currentLevel.add(infileNode);
+      Set<CompactionNode> nextLevel = new HashSet<>();
       int i = 1;
       while (currentLevel.size() != 0) {
-        LOG.warn("DAG Level :" + i++);
+        LOG.debug("DAG Level: " + i++);
         for (CompactionNode current : currentLevel) {
-          LOG.warn("acknowledging file " + current.fileName);
+          LOG.debug("Acknowledging file " + current.fileName);
           if (current.snapshotGeneration <= dest.snapshotGeneration) {
-            LOG.warn("Reached dest generation count. SrcSnapshot : " +
-                src.dbPath + " and Dest " + "Snapshot" + dest.dbPath +
-                " Contain Diffrent file " + current.fileName);
+            LOG.debug("Reached dest generation count. Src: " +
+                src.dbPath + " and Dest: " + dest.dbPath +
+                " have different file: " + current.fileName);
             differentFiles.add(current.fileName);
             continue;
           }
           Set<CompactionNode> successors = mutableGraph.successors(current);
-          if (successors == null || successors.size() == 0) {
-            LOG.warn("No further compaction for the file" +
-                ".SrcSnapshot : " + src.dbPath + " and Dest " +
-                "Snapshot" + dest.dbPath + " Contain Diffrent file " +
-                current.fileName);
+          if (successors.size() == 0) {
```

Comment:
> Is `successors` nullable?

Good point. But doesn't seem so according to Guava `Graph` [javadoc](https://www.javadoc.io/doc/com.google.guava/guava/31.1-jre/com/google/common/graph/Graph.html#successors(N)).

Suggestions other than this would been included in the next commit. Thx!